integration-tests/utils/integration-utils.js

```diff
@@ -171,8 +171,12 @@ const noSpinnersAfter = async (page, { action, debugMessage }) => {
   if (debugMessage) {
     console.log(`About to perform an action and wait for spinners. \n\tDebug message: ${debugMessage}`)
   }
-  const foundSpinner = page.waitForXPath('//*[@data-icon="loadingSpinner"]')
-  await Promise.all([foundSpinner, action()])
+  const foundSpinner = page.waitForXPath('//*[@data-icon="loadingSpinner"]', { timeout: 5000 })
```

Comment:
Display of loading spinner should not take 30 seconds. 5 seconds is enough time to wait for the loading spinner.